src/test/java/reposense/util/InputBuilder.java

```diff
@@ -0,0 +1,92 @@
+package reposense.util;
+
+import java.nio.file.Path;
+
+import reposense.parser.ArgsParser;
+
+/**
+ * A utility class to help with building command line input.
+ * Example usage: <br>
+ *     {@code String input = new InputBuilder().addSince("27/01/2017").build();}
+ */
+public class InputBuilder {
+    private static final String WHITESPACE = " ";
+
+    private String input;
+
+    public InputBuilder() {
+        this.input = "";
+    }
+
+    public String build() {
+        return input;
+    }
+
+    public InputBuilder addHelp() {
+        input += ArgsParser.HELP_FLAGS[0] + WHITESPACE;
+        return this;
+    }
+
+    public InputBuilder addConfig(Path path) {
+        input += ArgsParser.CONFIG_FLAGS[0] + WHITESPACE + path + WHITESPACE;
```

Comment:
Hmm, this actually allows us to add the config option multiple times (and for the other options below).
Maybe we can make `InputBuilder` hold a `boolean` flag to indicate if an option has been 'set', then only add it in when we call `build` if it has been 'set', else we don't

Reply:
I believe we are writing a test util class which will not be used in the main code. If there are misuses of this InputBuilder,  ParseException will be raised when executing the input. So I believe there is no need of making sure InputBuilder returns a valid string.

Reply:
Yes, this is a test util class. But by ensuring it gives a valid output, we minimize errors on our part (especially for newer developers), such as using the same `InputBuilder` to build 2 strings but the 2nd string giving an incorrect string, and ensures that any test failures that occurs is due to a bug in the main code, and not misusing our test code.

Reply:
I want to clarify one point. If a user adds the same flag multiple times, do you expect it to throw an exception, all just output the latest one that is added?

Reply:
If it throws an exception, do I need to write a new exception class for it?

Reply:
This behavior should be handled by the shell. So, I don't think our program would know if a flags is repeated. 🤔 

`i.e.` `./gradlew run -Dargs="-repo . -repo . -repo . -view -view -view"` or `ls -l -l -l -la -l`.

Also, we use Apache's `translateCommandline()` to mimic shell-like behavior in `ArgsParserTest`. Does it not work?

Reply:
>I want to clarify one point. If a user adds the same flag multiple times, do you expect it to throw an exception, all just output the latest one that is added?

Hmm, did some testing with `-config` option, seems like it will parse all the options, then give you the last 1 specified. However, I'm not sure if we want to add tests for this.
@yong24s @eugenepeh Your opinions on this?

Reply:
@fzdy1914 Anyway, since our `ArgsParser` allows for an option to be specified multiple times, we can keep the `InputBuilder` as it is now

Reply:
I was about to say that previous comment may not be actually correct. 

But running these on `Bash` yields different result than yours. 🤷‍♂️ 
`./gradlew run -Dargs="-repo a.git -repo b.git -repos c.git"` -> takes `c.git`
`./gradlew run -Dargs="-config a -config b"` -> takes `a`

If is behavior is caused by the shell then there's no point testing it?

Reply:
>If is behavior is caused by the shell then there's no point testing it?

Whoops, forgot to test with different options. But yea if it gives different results each time, then there's no point in testing this.
@fzdy1914 In this case, ignore my previous comment, we should ensure `InputBuilder` only allow adding of each option once. No need for another exception if user add flags more than once for now.

Reply:
How can we achieve this without using exceptions? I am not able to figure out an elegant way to achiece this effect.

Reply:
How can we achieve this without using exceptions? I am not able to figure out an elegant way to achiece this effect.

Reply:
How can we achieve this without using exceptions? I am not able to figure out an elegant way to achiece this effect.

Reply:
>How can we achieve this without using exceptions? I am not able to figure out an elegant way to achiece this effect.

Why do we need an exception for this? If we call 'set' for any option, we should just overwrite the existing value if it exists, ensure that each option can only be called once.

Reply:
I get your point. So we change add to set, correct?

Reply:
>I get your point. So we change add to set, correct?

Yes

Reply:
Hmm, 

>`./gradlew run -Dargs="-repo a.git -repo b.git -repos c.git"` -> takes `c.git`

Isn't that caused by the different in flag? e.g. `-repo` -> `-repos` may have caused the ArgParser to treat it differently.

I don't think terminal does anything to the args, they should be fed into the program as a whole string and parsed into string array to the main.
https://github.com/reposense/RepoSense/blob/f86a52617ee8ab13cdf696c0e196fdc90cf5f778/src/main/java/reposense/RepoSense.java#L34
![image](https://user-images.githubusercontent.com/19277206/52715339-32e73c80-2fd7-11e9-993f-c14a08bc5fb9.png)

Reply:
>`./gradlew run -Dargs="-config a -config b"` -> takes `a`

@yong24s Used 2 `-config` params and it took the 1st one, while when I used 2 `-config` params it took the last one instead, which is why we thought the behaviour is undefined/unpredictable depending on how to user uses it